### DISCUSSION TITLE
create-next-app should use the `src` folder by default

### DISCUSSION THREAD

**Discussion by Nsttt** [vercel/next.js#41839] [Ideas]
# Proposal
Currently when bootstraping a new app using `create-next-app` with the `--experimental-app` flag, the app bootstrap with the following structure.
![CleanShot 2022-10-26 at 10 56 36](https://user-images.githubusercontent.com/23436531/197982033-30edc069-b713-446a-89a4-057f01a92bac.png)

I think that with the new change to bootstrap using Typescript by default its a good opportunity to also bootstrap using the `src` folder convention by default.

This makes the folder structure cleaner by having all the code inside a root folder.

**Comment by t3dotgg** [comment]
Chiming in on behalf of create-t3-app, really think this is worth it 🙏

**Comment by eriicafes** [comment]
This will save me an extra commit to setup the src directory for all nextjs projects I didn’t use create-t3-app for 🙏❤️

**Comment by imNel** [comment]
+1! This 100% helps with file structure organisation/readability 😁

**Comment by jescalan** [comment]
Spoke with the team about this and I think the preferred route would be to have a separate flag for this, like `--experimental-app-src`. Would you be open to doing a PR for this? Could be a great contribution ☺️

**Reply by Nsttt** [reply]
Of course, would love to ❤️

**Reply by jescalan** [reply]
Amazing! Thank you so much 💖

Slight modification - thinking that `--src` would be a separate flag, so instead of `--experimental-app-src`, it would be `--experimental-app --src` - does that seem good? Would make it cover more use cases this way 😁

**Reply by Nsttt** [reply]
Absolutely, will try to make the PR over this week 😄

**Comment by parssak** [comment]
This would be amazing, +1. Always is the first thing I need to do before I can start using the template.
